llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -26658,6 +26670,18 @@ bool AArch64TargetLowering::SimplifyDemandedBitsForTargetNode(
     // used - simplify to just Val.
     return TLO.CombineTo(Op, ShiftR->getOperand(0));
   }
+  case AArch64ISD::BICi: {
+    // Fold BICi if all destination bits already known to be zeroed
+    SDValue Op0 = Op.getOperand(0);
+    KnownBits KnownOp0 = TLO.DAG.computeKnownBits(Op0, 0);
```

Comment:
Use the computeKnownBits variant with OriginalDemandedElts

Don't reset the recursion depth - it must be Depth + 1